src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/DateTimeValueResolver.php

```diff
@@ -48,30 +48,42 @@ public function resolve(Request $request, ArgumentMetadata $argument): iterable
 
         $class = \DateTimeInterface::class === $argument->getType() ? \DateTimeImmutable::class : $argument->getType();
         $format = null;
+        $timezone = null;
 
         if ($attributes = $argument->getAttributes(MapDateTime::class, ArgumentMetadata::IS_INSTANCEOF)) {
             $attribute = $attributes[0];
             $format = $attribute->format;
+            $timezone = $attribute->timezone;
         }
 
         $date = false;
+        $defaultTimezone = date_default_timezone_get();
+        if ('default' === $timezone) {
+            $timezone = $defaultTimezone;
+        } elseif ($timezone) {
+            date_default_timezone_set($timezone);
+        }
 
-        if (null !== $format) {
-            $date = $class::createFromFormat($format, $value);
+        try {
+            if (null !== $format) {
+                $date = $class::createFromFormat($format, $value);
```

Comment:
shouldn't the try/finally be replaced by adding a 3rd argument to createFromFormat?

Reply:
That would work for createFromFormat, but not for strtotime. I can refactor to uptade the global setting only when absolutely necessary.

Reply:
hum ok, let's keep as is then